iac-common/src/main/java/org/sonar/iac/common/yaml/YamlSensor.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.common.yaml;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.snakeyaml.engine.v2.exceptions.Mark;
+import org.snakeyaml.engine.v2.exceptions.MarkedYamlEngineException;
+import org.sonar.api.SonarRuntime;
+import org.sonar.api.batch.fs.FilePredicate;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.TextPointer;
+import org.sonar.api.batch.rule.CheckFactory;
+import org.sonar.api.batch.rule.Checks;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.issue.NoSonarFilter;
+import org.sonar.api.measures.FileLinesContextFactory;
+import org.sonar.api.resources.Language;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.common.extension.DurationStatistics;
+import org.sonar.iac.common.extension.IacSensor;
+import org.sonar.iac.common.extension.ParseException;
+import org.sonar.iac.common.extension.TreeParser;
+import org.sonar.iac.common.extension.visitors.ChecksVisitor;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.common.extension.visitors.TreeVisitor;
+import org.sonar.iac.common.yaml.visitors.YamlHighlightingVisitor;
+import org.sonar.iac.common.yaml.visitors.YamlMetricsVisitor;
+
+public abstract class YamlSensor extends IacSensor {
+
+  protected static final String JSON_LANGUAGE_KEY = "json";
+  protected static final String YAML_LANGUAGE_KEY = "yaml";
+  protected static final String FILE_SEPERATOR = "---";
+
+  protected final Checks<IacCheck> checks;
+  protected YamlSensor(SonarRuntime sonarRuntime, FileLinesContextFactory fileLinesContextFactory, CheckFactory checkFactory,
+                       NoSonarFilter noSonarFilter, Language language, List<Class<?>> checks) {
+    super(sonarRuntime, fileLinesContextFactory, noSonarFilter, language);
+    this.checks = checkFactory.create(repositoryKey());
+    this.checks.addAnnotatedChecks(checks);
+  }
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor
+      .onlyOnLanguages(JSON_LANGUAGE_KEY, YAML_LANGUAGE_KEY)
+      .name("IaC " + language.getName() + " Sensor");
+  }
+
+  @Override
+  protected TreeParser<Tree> treeParser() {
+    return new YamlParser();
+  }
+
+  @Override
+  protected List<TreeVisitor<InputFileContext>> visitors(SensorContext sensorContext, DurationStatistics statistics) {
+    List<TreeVisitor<InputFileContext>> visitors = new ArrayList<>();
+    visitors.add(new YamlHighlightingVisitor());
+    visitors.add(new YamlMetricsVisitor(fileLinesContextFactory, noSonarFilter));
```

Comment:
Previously these two visitors were only added if there was an activated SonarLint context. But maybe this has changed ?

Reply:
I removed the condition because SonarIac is not running in SonarLint and it's not planned yet.